x/exchange/keeper/commitments.go

```diff
@@ -426,3 +457,43 @@ func (k Keeper) consumeCommitmentSettlementFee(ctx sdk.Context, req *exchange.Ms
 	antewrapper.ConsumeMsgFee(ctx, exchangeFees.ExchangeFees, req, "")
 	return nil
 }
+
+// validateTransferCommitments which validate TransferCommitments message request.
+func (k Keeper) validateTransferCommitments(
+	ctx sdk.Context,
+	store storetypes.KVStore,
+	req *exchange.MsgMarketTransferCommitmentsRequest,
+	account sdk.AccAddress,
+) error {
+	// Blocked address check
+	if k.bankKeeper.BlockedAddr(account) {
+		return fmt.Errorf("%s is not allowed to receive funds", account)
+	}
+	// Zero amount
+	if req.Amount.IsZero() {
+		return fmt.Errorf("account %s does not have any funds committed to market %d", account, req.CurrentMarketId)
+	}
+
+	// Negative amount
+	if req.Amount.IsAnyNegative() {
+		return fmt.Errorf("cannot transfer negative commitment amount %q for %s in market %d", req.Amount, req.Account, req.CurrentMarketId)
+	}
+
+	// Market validity
+	if err := validateMarketIsAcceptingCommitments(store, req.CurrentMarketId); err != nil {
+		return fmt.Errorf("current market %d is invalid: %w", req.CurrentMarketId, err)
+	}
+	if err := validateMarketIsAcceptingCommitments(store, req.NewMarketId); err != nil {
+		return fmt.Errorf("new market %d is invalid: %w", req.NewMarketId, err)
+	}
+
+	// User permission
+	if err := k.validateUserCanCreateCommitment(ctx, req.CurrentMarketId, account); err != nil {
+		return fmt.Errorf("account %s does not have permission to create commitments in market %d: %w", account, req.CurrentMarketId, err)
+	}
```

Comment:
We're not creating a commitment in the current market, so we don't need to check this.

Use case: A user used to be able to commit funds to a market, but now, no longer has the required attributes, which is why they want their commitment transferred to a new market (that they're allowed to use).